### DISCUSSION TITLE
wire:click not working

### DISCUSSION THREAD

**Discussion by luisuran** [livewire/livewire#5132] [1. Help] [answered]
Hi. 
I have a component that I want to show/hide on clicking a div. I don't know what I'm doing wrong because it's not working.

This is where I want to click
```
<div wire:click="showTablaGestiones" id="collapseOne" class="panel-collapse collapse" role="tabpanel" aria-labelledby="headingOne">
      <div class="panel-body">
          {{-- @include('auth.ventas.panel.tabla',['gestiones' => $gestiones->where('estado_id', 5)->all()]) --}}
          <livewire:show-gestiones :gestiones="$gestiones" :cortes="$cortes" :estado=5 />
      </div>
    </div>
```

This is the controller
```
<?php

namespace App\Http\Livewire;

use Livewire\Component;

class ShowGestiones extends Component
{
    public $show;
    
    public function showTablaGestiones() {
        $this->show = !$this->show;
    }

    public function render()
    {
        return view('livewire.show-gestiones');
    }
}

```

This is the view
```
<span>
    @if ($show)
        Showing
    @else 
        Not showing
    @endif
</span>
```

It always returns "Not showing"

**Comment by joshhanley** [comment]
@luisuran it looks like you have your `wire:click` outside of your Livewire component. I needs to be inside the component for it to work properly.

Try changing that and see if it works 🙂

Hope this helps!

**Comment by denizozturk** [comment, marked as answer]
The "showTablaGestiones" method that you defined in the wire:click statement can be used in the view file of the "ShowGestiones" component. I can see you are trying to use it outside of the component. If the component has a livewire component in an upper layer, you can create a listen for the component inside and trigger it in the upper layer.

In addition, you can use the "action" example below for such toggle operations.

`wire:click="$toggle('show')"`

**Reply by luisuran** [reply]
Thank you very much. I can see now my obvious misconception. I'm just learning.
